adlfs/tests/test_spec.py

```diff
@@ -754,27 +753,51 @@ def test_rm_recursive(mock_delete_blob, storage):
         mock.ANY, "root/c"
     ), "The directory deletion should be the last call"
 
-
-@mock.patch.object(
-    azure.storage.blob.aio.ContainerClient,
-    "delete_blob",
-    side_effect=azure.storage.blob.aio.ContainerClient.delete_blob,
-    autospec=True,
-)
-def test_rm_recursive2(mock_delete_blob, storage):
+@pytest.mark.filterwarnings("error")
+def test_rm_recursive2(storage):
     fs = AzureBlobFileSystem(
         account_name=storage.account_name, connection_string=CONN_STR
     )
 
     assert "data/root" in fs.ls("/data")
 
     fs.rm("data/root", recursive=True)
+
     assert "data/root" not in fs.ls("/data")
 
     with pytest.raises(FileNotFoundError):
         fs.ls("data/root")
 
-    last_deleted_paths = [call.args[1] for call in mock_delete_blob.mock_calls[-7:]]
+
+async def test_rm_recursive_call_order(storage, mocker):
```

Comment:
Since `fs.rm` validates that the files are actually removed, there wasn't an easy way to mock the `delete_blob` so that still ran the deletes asynchronously as a side effect (someone might have a clever way to do this). So instead I just explicitly supply the file paths under `data/root` , call `fs._rm_files`, and assert the call order similar to how the previous test was working